request-management-api/request_api/resources/foidocument.py

```diff
@@ -103,6 +104,45 @@ def post(requestid, documentid, requesttype):
         except BusinessException as exception:            
             return {'status': exception.status_code, 'message':exception.message}, 500 
         
+@cors_preflight('POST,OPTIONS')
+@API.route('/foidocument/<requesttype>/<requestid>/documentid/<documentid>/reclassify')
+class ReclassifyFOIDocument(Resource):
+    """Resource for reclassifying uploaded attachments of FOI requests."""
+
+    @staticmethod
+    @TRACER.trace()
+    @cross_origin(origins=allowedorigins())
+    @auth.require
+    def post(requesttype, requestid, documentid):
+        try:
+            requestjson = request.get_json()
+            activedocuments = documentservice().getactiverequestdocuments(requestid, requesttype)
+            documentpath = 'no documentpath found'
+            if (requesttype == 'ministryrequest'):
+                for document in activedocuments:
+                    if document['foiministrydocumentid'] == int(documentid):
+                        documentpath = document['documentpath']
+            else:
+                for document in activedocuments:
+                    if document['foidocumentid'] == int(documentid):
+                        documentpath = document['documentpath']
+
+            # move document in S3
+            moveresult = documentservice().copyrequestdocumenttonewlocation(requestjson['category'], documentpath)
+            # save new version of document with updated documentpath
+            updatedata = {'category': requestjson['category'], 'documentpath': moveresult['documentpath']}
+            documentschema = ReclassifyDocumentSchema().load(updatedata)
+            if moveresult['status'] == 'success':
+                 result = documentservice().createrequestdocumentversion(requestid, documentid, documentschema, AuthHelper.getuserid(), requesttype)
+                 return {'status': result.success, 'message':result.message,'id':result.identifier} , 200
+            return {'status': False, 'message': 'Something went wrong' }, 500
+        except ValidationError as err:
+                    return {'status': False, 'message':err.messages}, 400
+        except KeyError as err:
+            return {'status': False, 'message': 'KeyError'}, 400
+        except BusinessException as exception:
+            return {'status': exception.status_code, 'message':exception.message}, 500
+
```

Comment:
Please review best way to handle errors for this endpoint. 
When the KeyError exception message was set to to err.messages, it was giving a "KeyError does not have messages attribute" error.